doc/running/caching.rst

```diff
@@ -0,0 +1,52 @@
+Caching in mirgecom
+===================
+
+
+OpenCL Kernels
+--------------
+
+OpenCL kernels are cached on hard disks on multiple levels in a mirgecom
+execution. This has the advantage of reducing the compilation time of kernels
+when running the same driver multiple times, but can lead to inconsistent
+performance results, especially when comparing multiple executions against
+each other, as subsequent executions can appear faster than earlier ones due
+to caching effects.
+
+
+The following list discusses mirgecom-related packages that use caching.
+
+Loopy
++++++
+
+Loopy stores generated PyOpenCL kernels in ``$XDG_CACHE_HOME`` by default. You can
+export ``LOOPY_NO_CACHE=1`` to disable caching. See
+`here <https://github.com/inducer/loopy/blob/e21e8f85d289abbca27ac6abfd71874155fa49da/loopy/__init__.py#L402-L406>`__
+for details.
```

Comment:
To be technically correct, Loopy has a few more layers of cache:
```
$ ls .cache/pytools
pdict-v4-loopy-buffer-array-cache-2021.1-islpy2020.2.2-cgen2020.2-5b84d6d9e0ad50901771ceb7314ee6097dd18399-v1-py3.9.2.final.0
pdict-v4-loopy-buffer-array-cache-2021.1-islpy2020.2.2-cgen2020.2-c5e212fc440759cc4074fbf760dcc97dad5fe428-v1-py3.9.2.final.0
pdict-v4-loopy-code-gen-cache-v3-2021.1-islpy2020.2.2-cgen2020.2-5b84d6d9e0ad50901771ceb7314ee6097dd18399-v1-py3.9.2.final.0
pdict-v4-loopy-code-gen-cache-v3-2021.1-islpy2020.2.2-cgen2020.2-c5e212fc440759cc4074fbf760dcc97dad5fe428-v1-py3.9.2.final.0
pdict-v4-loopy-invoker-cache-v1-2021.1-islpy2020.2.2-cgen2020.2-5b84d6d9e0ad50901771ceb7314ee6097dd18399-v1-py3.9.2.final.0
pdict-v4-loopy-invoker-cache-v1-2021.1-islpy2020.2.2-cgen2020.2-c5e212fc440759cc4074fbf760dcc97dad5fe428-v1-py3.9.2.final.0
pdict-v4-loopy-preprocess-cache-v2-2021.1-islpy2020.2.2-cgen2020.2-5b84d6d9e0ad50901771ceb7314ee6097dd18399-v1-py3.9.2.final.0
pdict-v4-loopy-preprocess-cache-v2-2021.1-islpy2020.2.2-cgen2020.2-c5e212fc440759cc4074fbf760dcc97dad5fe428-v1-py3.9.2.final.0
pdict-v4-loopy-schedule-cache-v4-2021.1-islpy2020.2.2-cgen2020.2-5b84d6d9e0ad50901771ceb7314ee6097dd18399-v1-py3.9.2.final.0
pdict-v4-loopy-schedule-cache-v4-2021.1-islpy2020.2.2-cgen2020.2-c5e212fc440759cc4074fbf760dcc97dad5fe428-v1-py3.9.2.final.0
pdict-v4-loopy-typed-and-scheduled-cache-v1-2021.1-islpy2020.2.2-cgen2020.2-5b84d6d9e0ad50901771ceb7314ee6097dd18399-v1-py3.9.2.final.0
pdict-v4-loopy-typed-and-scheduled-cache-v1-2021.1-islpy2020.2.2-cgen2020.2-c5e212fc440759cc4074fbf760dcc97dad5fe428-v1-py3.9.2.final.0
pdict-v4-pyopencl-generated-scan-kernel-cache-v1-py3.9.2.final.0
pdict-v4-pyopencl-invoker-cache-v38-py3.9.2.final.0
pdict-v4-sumpy-code-cache-v6-2020.2beta1-py3.9.2.final.0
```

Reply:
I expanded the description a bit here, but imho we don't need to go much into detail here.